docs/design/rule-manager/README.md

```diff
@@ -0,0 +1,75 @@
+# Rule manager
+
+## Overview
+
+Rule manager is a separate open-kilda component responsible for building rules. The main goal of this component is to encapsulate all rule related logic into one place. All other components (flow CRUD, flow/switch validation and sync, default rules management, etc) should use rule manager API to build rules.
+
+## Details
+
+Rule manager implemented as a separate library module. It should support input data from database (for internal system calls) and data provided by user in json format (for manual calls). Manual calls should have CLI options to support data input from a file. Rule manager should use adapters to encapsulate input data and provide the same results for different input types. 
+
+The main idea is to build rules as close as possible to the actual OpenFlow representation so Floodlight may just translate messages directly into switch commands. Rule manager doesn't send any commands to anywhere it's responsible only for building rules. Sending results to floodlight is out of scope and should be covered by calling component.
```

Comment:
OF flow will differ for different OF version. We should define more preciselly "rule" representation. Do we want to know these differences inside this manager?

I would start from desing of speaker protocol, because it is precondition for this manager.

Reply:
Added OF version into Rule Manager responsibilities. Please check speaker protocol in Json format in example.